twilight-model/src/gateway/payload/incoming/member_add.rs

```diff
@@ -1,79 +1,88 @@
-use crate::guild::Member;
+use crate::{
+    guild::Member,
+    id::{marker::GuildMarker, Id},
+};
 use serde::{Deserialize, Serialize};
 use std::ops::{Deref, DerefMut};
 
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct MemberAdd(pub Member);
+pub struct MemberAdd {
+    pub guild_id: Id<GuildMarker>,
+    #[serde(flatten)]
+    pub member: Member,
+}
 
 impl Deref for MemberAdd {
     type Target = Member;
 
     fn deref(&self) -> &Self::Target {
-        &self.0
+        &self.member
     }
 }
 
 impl DerefMut for MemberAdd {
     fn deref_mut(&mut self) -> &mut Self::Target {
-        &mut self.0
+        &mut self.member
     }
 }
```

Comment:
I think this is confusing and I don't think we do this for any other struct with multiple items. Should it really be preserved? cc @zeylahellyer